catroid/src/org/catrobat/catroid/content/Project.java

```diff
@@ -136,6 +136,39 @@ public XmlHeader getXmlHeader() {
 		return this.xmlHeader;
 	}
 
+	public List<String> getRequiredResources() {
+		int resources = Brick.NO_RESOURCES;
+
+		for (Sprite sprite : spriteList) {
+			resources |= sprite.getRequiredResources();
+		}
```

Comment:
https://github.com/Catrobat/Catroid/blob/master/catroid/src/org/catrobat/catroid/stage/PreStageActivity.java#L313-L322
Looks like the same code to me. Please refactor

Btw: Not sure why the PreStageActivity has a member called resources. Maybe it's needed in the tests... https://github.com/Catrobat/Catroid/blob/master/catroid/src/org/catrobat/catroid/stage/PreStageActivity.java#L75